internal/service/source.go

```diff
@@ -188,3 +192,117 @@ func (h *ServiceHandler) UpdateSource(ctx context.Context, request server.Update
 
 	return server.UpdateSource200JSONResponse{}, nil
 }
+
+func (h *ServiceHandler) UploadRvtoolsFile(ctx context.Context, request server.UploadRvtoolsFileRequestObject) (server.UploadRvtoolsFileResponseObject, error) {
+    multipartReader := request.Body
+    var rvtoolsContent []byte
+    
+    for {
+        part, err := multipartReader.NextPart()
+        if err == io.EOF {
+            break
+        }
+        if err != nil {
+            return server.UploadRvtoolsFile400JSONResponse{
+                Message: "Failed to read multipart data",
+            }, nil
+        }
+        
+        if part.FormName() == "file" {
+            rvtoolsContent, err = io.ReadAll(part)
+            if err != nil {
+                return server.UploadRvtoolsFile400JSONResponse{
+                    Message: "Failed to read uploaded file content",
+                }, nil
+            }
+            break
+        }
+    }
+    
+    if rvtoolsContent == nil {
+        return server.UploadRvtoolsFile400JSONResponse{
+            Message: "No file was found in the request",
+        }, nil
+    }
+    
+    if len(rvtoolsContent) == 0 {
+        return server.UploadRvtoolsFile400JSONResponse{
+            Message: "Empty file uploaded",
+        }, nil
+    }
+	zap.S().Infof("Received RVTools data with size: %d bytes", len(rvtoolsContent))
+    
+	//TODO: support csv files
+    if !rvtools.IsExcelFile(rvtoolsContent) {
+        return server.UploadRvtoolsFile400JSONResponse{
+            Message: "The uploaded file is not a valid Excel (.xlsx) file. Please upload an RVTools export in Excel format.",
+        }, nil
+    }
+    
+    inventory, err := rvtools.ParseRVTools(rvtoolsContent)
+    if err != nil {
+        return server.UploadRvtoolsFile400JSONResponse{
+            Message: fmt.Sprintf("Error parsing RVTools file: %v", err),
+        }, nil
+    }
+
+    source, err := h.store.Source().Get(ctx, request.Id)
+    if err != nil {
+        return server.UploadRvtoolsFile400JSONResponse{
+            Message: "Failed to retrieve source",
+        }, nil
+    }
+
+    if source == nil {
```

Comment:
I think this is redudant, because if resource is not found in DB, we return error